model/DataObject.php

```diff
@@ -1782,6 +1804,11 @@ public function getManyManyComponents($componentName, $filter = null, $sort = nu
 
 		$extraFields = $this->manyManyExtraFieldsForComponent($componentName) ?: array();
 		$result = ManyManyList::create($componentClass, $table, $componentField, $parentField, $extraFields);
+
+		// Store component data in query meta-data
+		$result = $result->alterDataQuery(function($query) use ($extraFields) {
+			$query->setQueryParam('Component.ExtraFields', $extraFields);
+		});
```

Comment:
This is, as far as I can see, the most efficient way of ensuring we have access to this info in `castingHelper()`. No extra lookups/singletons are needed. Wasn’t really sure what to name the parameter though...